### PR TITLE
Bug 1705627: Add --show-console flag to whoami command

### PR DESCRIPTION
**Pull request by mfojtik**
This flag will fetch configmap that contain location of web console URL and print it to user.
For 3.x API servers, this will use the API server hostname which should redirect user to console.

Example output:

```
$ oc whoami --show-console
https://api.ci.openshift.org:443
```

xref: https://github.com/openshift/console-operator/pull/218

### CODE REVIEW

**Comment by smarterclayton** [comment]
Add example output to description.  Also link bug in PR title :)

File: pkg/oc/cli/whoami/whoami.go [hunk 1]
```diff
@@ -4,19 +4,25 @@ import (
...
 	"k8s.io/kubernetes/pkg/kubectl/util/templates"
 
 	userv1 "github.com/openshift/api/user/v1"
 	userv1typedclient "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
 )
 
-const WhoAmIRecommendedCommandName = "whoami"
+const (
+	WhoAmIRecommendedCommandName        = "whoami"
+	OpenShiftConfigManagedNamespaceName = "openshift-config-managed"
```
Comment (smarterclayton) [inline pkg/oc/cli/whoami/whoami.go]: private constant

File: pkg/oc/cli/whoami/whoami.go [hunk 2]
```diff
@@ -4,19 +4,25 @@ import (
...
+	ConsolePublicConfigMap              = "console-public"
```
Comment (smarterclayton) [inline pkg/oc/cli/whoami/whoami.go]: private constant

File: pkg/oc/cli/whoami/whoami.go [hunk 3]
```diff
@@ -96,18 +112,33 @@ func (o *WhoAmIOptions) Validate() error {
...
+	if errors.IsNotFound(err) {
+		return fmt.Sprintf("https://", o.ClientConfig.Host), nil
+	}
+	if err != nil {
+		return "", fmt.Errorf("unable to determine web console URL: %v", err)
+	}
+
+	consoleUrl, exists := consolePublicConfig.Data["consoleURL"]
+	if !exists {
+		return "", fmt.Errorf("unable to determine web console URL: 'consoleURL' key not found in %s/%s configmap", OpenShiftConfigManagedNamespaceName, ConsolePublicConfigMap)
```
Comment (smarterclayton) [inline pkg/oc/cli/whoami/whoami.go]: this error is a bit wordy.  `unable to determine console location from the cluster` is sufficient

File: test/cmd/whoami.sh [hunk 4]
```diff
@@ -13,6 +13,7 @@ trap os::test::junit::reconcile_output EXIT
 os::test::junit::declare_suite_start "cmd/whoami"
 # This test validates the whoami command's --show-server flag
 os::cmd::expect_success_and_text 'oc whoami --show-server' 'http(s)?:\/\/localhost\:[0-9]+'
+os::cmd::expect_success_and_text 'oc whoami --show-console' 'https://'
```
Comment (smarterclayton) [inline test/cmd/whoami.sh]: it would be better to do `"https://$( read from config the value of host )"`

File: pkg/oc/cli/whoami/whoami.go [hunk 5]
```diff
@@ -76,11 +85,18 @@ func (o WhoAmIOptions) WhoAmI() (*userv1.User, error) {
 
 func (o *WhoAmIOptions) Complete(f kcmdutil.Factory) error {
 	var err error
+
 	o.ClientConfig, err = f.ToRESTConfig()
 	if err != nil {
 		return err
 	}
 
+	kubeClient, err := kubernetes.NewForConfig(o.ClientConfig)
```
Comment (soltysh) [inline pkg/oc/cli/whoami/whoami.go]: Assign directly to `o.KubeClient`.

File: test/cmd/whoami.sh [hunk 6]
```diff
@@ -13,6 +13,7 @@ trap os::test::junit::reconcile_output EXIT
 os::test::junit::declare_suite_start "cmd/whoami"
 # This test validates the whoami command's --show-server flag
 os::cmd::expect_success_and_text 'oc whoami --show-server' 'http(s)?:\/\/localhost\:[0-9]+'
+os::cmd::expect_success_and_text 'oc whoami --show-console' 'http(s)?:\/\/localhost\:[0-9]+'
```
Comment (soltysh) [inline test/cmd/whoami.sh]: Would be e2e worthy, although that might happen when I move all of the tests to a real cluster.
